cmake/onnxruntime_unittests.cmake

```diff
@@ -886,7 +886,7 @@ if (onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
 endif()
 if (CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
   set_target_properties(onnxruntime_test_all PROPERTIES LINK_DEPENDS ${TEST_SRC_DIR}/wasm/onnxruntime_test_all_adapter.js)
-  set_target_properties(onnxruntime_test_all PROPERTIES LINK_FLAGS "-s STACK_SIZE=5242880 -s ALLOW_MEMORY_GROWTH=1 --pre-js \"${TEST_SRC_DIR}/wasm/onnxruntime_test_all_adapter.js\" -s \"EXPORTED_RUNTIME_METHODS=['FS']\" --preload-file ${CMAKE_CURRENT_BINARY_DIR}/testdata@/testdata -s EXIT_RUNTIME=1 -s DEMANGLE_SUPPORT=1")
+  set_target_properties(onnxruntime_test_all PROPERTIES LINK_FLAGS "-s STACK_SIZE=5242880 -s ALLOW_MEMORY_GROWTH=1 --pre-js \"${TEST_SRC_DIR}/wasm/onnxruntime_test_all_adapter.js\" -s \"EXPORTED_RUNTIME_METHODS=['FS']\" -s INCOMING_MODULE_JS_API=[preRun] --preload-file ${CMAKE_CURRENT_BINARY_DIR}/testdata@/testdata -s EXIT_RUNTIME=1 -s DEMANGLE_SUPPORT=1")
```

Comment:
Should it use https://cmake.org/cmake/help/latest/command/target_link_options.html instead?

I think here we want to append new flags there instead of just set and replace the existing flags?

Reply:
You can try:
```
target_link_options(onnxruntime_test_all PRIVATE "-s STACK_SIZE=5242880")
target_link_options(onnxruntime_test_all PRIVATE "-s ALLOW_MEMORY_GROWTH=1")
...
```

or 
```
target_link_options(onnxruntime_test_all PRIVATE "LINKER:-s STACK_SIZE=5242880")
target_link_options(onnxruntime_test_all PRIVATE "LINKER:-s ALLOW_MEMORY_GROWTH=1")
...
```

or 
```
target_link_options(onnxruntime_test_all PRIVATE "LINKER:SHELL:-s STACK_SIZE=5242880")
target_link_options(onnxruntime_test_all PRIVATE "LINKER:SHELL:-s ALLOW_MEMORY_GROWTH=1")
...
```